samples/webforms/embed-a-dashboard-into-an-app/PBIWebApp/Site.Master

```diff
@@ -0,0 +1,103 @@
+﻿<%@ Master Language="C#" AutoEventWireup="true" CodeBehind="Site.master.cs" Inherits="PBIWebApp.SiteMaster" %>
+
+<!DOCTYPE html>
+
+<html lang="en">
+<head runat="server">
+    <title>Power BI - Web application</title>
+    <script type="text/javascript">
+
+        window.onload = function () {
+            // client side click to embed a selected dashboard.
+            var el = document.getElementById("bEmbedDashboardAction");
+                if (el.addEventListener) {
+                    el.addEventListener("click", updateEmbedDashboard, false);
+                } else {
+                    el.attachEvent('onclick', updateEmbedDashboard);
+                }
+
+            // listen for message to receive message from embedded dashboard.
+            if (window.addEventListener) {
+                window.addEventListener("message", receiveMessage, false);
+            } else {
+                window.attachEvent("onmessage", receiveMessage);
+            }
+
+            // handle server side post backs, optimize for reload scenarios
+            // show embedded dashboard if all fields were filled in.
+            var accessTokenElement = document.getElementById('MainContent_accessTokenTextbox');
+            if(null !== accessTokenElement){
+                var accessToken = accessTokenElement.value;
+                if ("" !== accessToken)
+                    updateEmbedDashboard();
+            }
+        };
+
+        // The embedded dashboard posts message for errors to parent window.  listen and handle as appropriate
+        function receiveMessage(event)
+        {
+            if (event.data) {
+                try {
+                    messageData = JSON.parse(event.data);
+                    if (messageData.event === "dashboardPageLoaded")
+                    {
```

Comment:
This event isn't sent currently (it is sent only when loading a dashboard using the sdk). However, the tileClick event is being sent, so I will handle it instead of the pageLoaded event.